src/network/server.rs

```diff
@@ -737,7 +738,36 @@ impl HubService for MyHubService {
 
                         for event in old_events.events {
                             if event_types.contains(&event.r#type) {
-                                let event = Self::rewrite_hub_event(event, store.shard_id);
+                                let mut event = Self::rewrite_hub_event(event, store.shard_id);
+
+                                if last_chunk
+                                    .as_ref()
+                                    .map(|chunk| {
+                                        return event.block_number
+                                            != chunk
+                                                .header
+                                                .as_ref()
+                                                .unwrap()
+                                                .height
+                                                .unwrap()
+                                                .block_number;
+                                    })
+                                    .unwrap_or(true)
+                                {
+                                    let chunk = store.shard_store.get_chunk_by_height(
+                                        Height {
+                                            shard_index: store.shard_id,
+                                            block_number: event.block_number,
+                                        }
+                                        .as_u64(),
+                                    );
+                                    last_chunk = chunk.unwrap_or(None);
+                                }
+
+                                if let Some(chunk) = &last_chunk {
+                                    event.produced_at = chunk.header.as_ref().unwrap().timestamp;
```

Comment:
Hmm, I think this should go into rewrite_hub_event. It's best to have a single place where the events are mutated. Otherwise we're going to miss things.